lib/api/core/validation/baseType.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @constructor
+ */
+function BaseType () {
+
+}
+
+/**
+ * Must be overrided in order for the validation type to be useful
+ *
+ * @return {boolean}
+ */
+BaseType.prototype.validate = function () {
+  return true;
+};
+
+/**
+ * Must be overrided if the type implements specific type_options
+ *
+ * @return {boolean}
+ * @this IpAddressType
+ */
+BaseType.prototype.validateFieldSpecification = function () {
+  return true;
+};
+
+
+/**
+ * @param {*} object
+ * @param {string[]} allowedProperties
+ * @returns {boolean}
+ */
+BaseType.prototype.checkAllowedProperties = function (object, allowedProperties) {
+  var result = true;
+
+  if (typeof object !== 'object') {
+    return false;
+  }
+
+  Object.keys(object).forEach(propertyName => {
```

Comment:
This would be better written usinng Array.every:

```js
return Object.keys(object).every(p => allowedProperties.indexOf(p) !== -1);
```

Array.every() stops at the first falsey callback, while Array.forEach() executes it for all array elements, regardless of its return value.